aiida/cmdline/params/options/commands/setup.py

```diff
@@ -59,8 +69,7 @@ def get_repository_path_default(ctx):
     import os
     from aiida.manage.configuration.settings import AIIDA_CONFIG_FOLDER
 
-    if ctx.params['profile'] is None or not isinstance(ctx.params['profile'], Profile):
-        raise click.BadParameter('specifying the name of the profile is required', param_hint='"--profile"')
+    validate_profile_parameter(ctx)
```

Comment:
since you are repeating yourself here with validating in both places, just wondering whether this is necessary (maybe yes, you let me know).
Otherwise all good!

Reply:
Yes this is necessary, because these functions can be called independently. Although reading the code a bit more carefully now, there is a bug in the other function I think. It is supposed to return the default if the profile is not specified, but now it will always throw the `BadParameter` exception